source/monitoring/cluster-monitoring.txt

```diff
@@ -0,0 +1,249 @@
+   :start-after: start-available-subscriber
+   :end-before: end-available-subscriber
+   :language: ruby
+   :copyable:
+   :dedent:
+
+The following table provides available subscribers and their monitoring ``topic``: 
+
+.. list-table::
+   :header-rows: 1
+   :widths: 35 20 45
+
+   * - Method Name
+     - Monitoring Topic
+     - Description 
+
+   * - `SDAMLogSubscriber <{+api-root+}/Mongo/Monitoring/SDAMLogSubscriber.html>`__
+     - 
+     - Subscribes to SDAM events and logs them.
+
+   * - `ServerClosedLogSubscriber <{+api-root+}/Mongo/Monitoring/ServerClosedLogSubscriber.html>`__
+     - ``SERVER_CLOSED``
+     - Subscribes to ``ServerClosed`` events and logs them.
+
+   * - `ServerDescriptionChangedLogSubscriber <{+api-root+}/Mongo/Monitoring/ServerDescriptionChangedLogSubscriber.html>`__
+     - ``SERVER_DESCRIPTION_CHANGED``
+     - Subscribes to ``ServerDescriptionChanged`` events and logs them.
+
+   * - `ServerOpeningLogSubscriber <{+api-root+}/Mongo/Monitoring/ServerOpeningLogSubscriber.html>`__ 
+     - ``SERVER_OPENING``
+     - Subscribes to ``ServerOpening`` events and logs them.
+
+   * - `TopologyChangedLogSubscriber <{+api-root+}/Mongo/Monitoring/TopologyChangedLogSubscriber.html>`__
+     - ``TOPOLOGY_CHANGED``
+     - Subscribes to ``TopologyChanged`` events and logs them.
+
+   * - `TopologyClosedLogSubscriber <{+api-root+}/Mongo/Monitoring/TopologyClosedLogSubscriber.html>`__
+     - ``TOPOLOGY_CLOSED``
+     - Subscribes to ``TopologyClosed`` events and logs them.
+
+   * - `TopologyOpeningLogSubscriber <{+api-root+}/Mongo/Monitoring/TopologyOpeningLogSubscriber.html>`__
+     - ``TOPOLOGY_OPENING``
+     - Subscribes to ``TopologyOpening`` events and logs them.
+
+You can find a table of SDAM event descriptions in the :ref:`Event Descriptions <SDAM-event-descriptions>` 
+section on this page. 
+
+Custom SDAM Monitoring
+----------------------
+
+You can create a custom SDAM subscriber to access details about server
+and topology events. Create a separate class for each event type, as available
+data for each event varies. 
+
+For all events, the ``succeeded`` method is called on each event subscriber with 
+the event as the sole argument. A simple SDAM logging subscriber can look 
+like the following:
+
+.. literalinclude:: /includes/monitoring/monitoring.rb
+   :start-after: start-sdam
+   :end-before: end-sdam
+   :language: ruby
+   :copyable:
+   :dedent:
+
+To subscribe to events, you need to create the appropriate subscriber and 
+subscribe to the correct monitoring ``topic``. The following code shows how 
+to subscribe to SDAM events globally:
+
+.. literalinclude:: /includes/monitoring/monitoring.rb
+   :start-after: start-sdam-subscriber-global
+   :end-before: end-sdam-subscriber-global
+   :language: ruby
+   :copyable:
+   :dedent:
+
+The following code shows how to subscribe to SDAM events for a single client 
+by using the ``sdam-proc`` client option:
```

Comment:
Q: does the first code example that subscribes on the client level also need the sdam-proc option?

Reply:
No, the ``Proc`` object is used to pass multiple subscribe commands at once to the client! The first example only subscribes to one event, so it is not necessary.